### PR TITLE
Accessibility updates

### LINKED ISSUE #45: Accessibility: focus the modal when it's brought up on the screen
We can probably do it the same way it's been done in o-tabs:

1. Set a `tabindex="0"` on the panel: https://github.com/Financial-Times/o-tabs/blob/master/src/js/Tabs.js#L34
2. Focus and set the outline to 0: https://github.com/Financial-Times/o-tabs/blob/master/src/js/Tabs.js#L62

### LINKED ISSUE #48: Accessibility: aria attributes
### On the button

```
aria-pressed="true/false" # depending on the state of the popup
aria-haspopup="true"
```

When triggered with an element that's not a button, should also have:
```
tabindex="0"
role="button" 
```

### On the overlay

```
role="dialog"
aria-labelledby="{ID of the heading}"
```

### LINKED ISSUE #54: Accessibility: restore initial focus when leaving the dialog
See https://www.marcozehe.de/2015/02/05/advanced-aria-tip-2-accessible-modal-dialogs/:

> Of course, Escape isn’t the only way to close your dialog. You would also provide a button to accept changes and execute on whatever functionality you desire. In any case, after you close the dialog, you must!!!, you absolutely must!!! then set focus back to the element that opened the dialog, or any other useful element from which the user of your app will most likely continue working.

### PR DESCRIPTION
**Pull request by onishiweb**
Leaving this here for comment on the solutions for the time being. 

There's some work to do on getting the correct testing set up in other PRs which allow the switch to ES6, so will be doing that first.

Fixes #45,
Fixes #48,  and
Fixes #54

### CODE REVIEW

**Comment by onishiweb** [comment]
@AlbertoElias this is ready to review now

File: src/js/overlay.js [hunk 1]
```diff
@@ -125,6 +125,10 @@ const Overlay = function(id, opts) {
 };
 
 Overlay.prototype.open = function() {
+	if(this.opts.trigger) {
```
Comment (AlbertoElias) [inline src/js/overlay.js]: for consistency, we usually add a space between if and the (

File: src/js/overlay.js [hunk 2]
```diff
@@ -159,6 +163,10 @@ Overlay.prototype.render = function() {
 	const wrapperEl = document.createElement('div');
 	wrapperEl.className = 'o-overlay';
 	wrapperEl.classList.add('o-overlay--' + this.id.replace(' ', '-'));
+	wrapperEl.setAttribute('role', 'dialog');
+	if(this.opts.heading) {
+		wrapperEl.setAttribute('aria-labelledby', this.opts.heading.title);
```
Comment (AlbertoElias) [inline src/js/overlay.js]: Isn't this supposed to be an id?

Comment (onishiweb) [inline src/js/overlay.js]: Do you mean the labelled-by or the heading?

Comment (onishiweb) [inline src/js/overlay.js]: Yep, you're correct. After lunch I'll check what's best to be used in this situation and update it.

**Comment by onishiweb** [comment]
@AlbertoElias Fixed those issues, ready to review again

File: src/js/overlay.js [hunk 3]
```diff
@@ -166,7 +171,10 @@ Overlay.prototype.render = function() {
 
 	if (this.opts.heading) {
 		const heading = document.createElement('header');
+		const headingId = this.opts.heading.title.replace(' ', '-').toLowerCase();
```
Comment (AlbertoElias) [inline src/js/overlay.js]: missing a test for this, rest looks great!

Comment (AlbertoElias) [inline src/js/overlay.js]: I guess you'd need a regex for that, don't think it's that important no, feel free to merge!

**Pull request merged**